src/sharding.js

```diff
@@ -100,6 +100,9 @@ class ShardingDatastore extends Adapter {
    */
   static async create (store, shard) {
     const hasShard = await store.has(shardKey)
+    if (!hasShard && !shard) {
+      throw new Error('Shard is required when datastore doesn\'t have a shard key already.')
```

Comment:
Should be a [dbOpenFailedError](https://github.com/ipfs/interface-datastore/blob/master/src/errors.js#L8)?